src/common/libjob/jj.c

```diff
@@ -126,9 +127,8 @@ int jj_get_counts_json (json_t *jobspec, struct jj_counts *jj)
     }
     if (version != 1) {
         snprintf (jj->error, sizeof (jj->error) - 1,
-                 "Invalid version: expected 1, got %d", version);
-        errno = EINVAL;
-        return -1;
+                 "Unsupported version: expected 1, got %d", version);
+        flux_log(NULL, LOG_WARNING, "%s: %s", __FUNCTION__, jj->error);
     }
```

Comment:
The warning will be emitted on stderr, which may not be appropriate in all cases, so  I think the entire conditional block should just be removed here.   Maybe a comment referencing the issue would be useful though,
like
```c
// jobspec version check omitted as discussed in #6632 and #6682
```